scipy/integrate/_tanhsinh.py

```diff
@@ -915,17 +919,18 @@ def _nsum_iv(f, a, b, step, args, log, maxterms, atol, rtol):
     return f, a, b, step, valid_abstep, args, log, maxterms_int, atol, rtol
 
 
-def _nsum(f, a, b, step=1, args=(), log=False, maxterms=int(2**20), atol=None,
-          rtol=None):
-    r"""Evaluate a convergent sum.
+def nsum(f, a, b, *, step=1, args=(), log=False, maxterms=int(2**20), atol=None,
+         rtol=None):
```

Comment:
```suggestion
def nsum(f, a, b, *, step=1, args=(), log=False, maxterms=int(2**20), tolerances=None):
```
Maybe should do this like in gh-20800.

Reply:
I agree with the rename.

Reply:
Done (locally).